testing/trino-product-tests/src/main/java/io/trino/tests/product/hudi/TestHudiSparkCompatibility.java

```diff
@@ -320,6 +325,48 @@ public void testTimelineTableRedirect()
         }
     }
 
+    @Test(groups = {HUDI, PROFILE_SPECIFIC_TESTS})
+    public void testTimeTravelQuery()
+    {
+        String tableName = "test_hudi_cow_select_session_props" + randomNameSuffix();
+        SimpleDateFormat formatter = new SimpleDateFormat(DEFAULT_INSTANT_TIME_FORMAT);
+        formatter.setTimeZone(TimeZone.getTimeZone(ZoneOffset.UTC));
+        String before = formatter.format(new Date());
+        onHudi().executeQuery("SET hoodie.schema.on.read.enable=true");
+        createNonPartitionedTable(tableName, COW_TABLE_TYPE);
+        String after = formatter.format(new Date());
+
+        try {
+            assertThat(onTrino().executeQuery("SELECT id, name FROM hudi.default." + tableName + " FOR VERSION AS OF " + before)).hasNoRows();
+            assertThat(onTrino().executeQuery("SELECT id, name FROM hudi.default." + tableName + " FOR VERSION AS OF " + after))
+                    .containsOnly(ImmutableList.of(
+                            row(1, "a1"),
+                            row(2, "a2")));
+
+            onHudi().executeQuery("ALTER TABLE default." + tableName + " ADD COLUMNS (new_col string)");
+            onHudi().executeQuery("INSERT INTO default." + tableName + " VALUES (3, 'a3', 20, 1000, 'a3_commit')");
+            String afterAddColumn = formatter.format(new Date());
+            assertThat(onTrino().executeQuery("SELECT id, name, new_col FROM hudi.default." + tableName + " FOR VERSION AS OF " + afterAddColumn))
+                    .containsOnly(ImmutableList.of(
+                            row(1, "a1", null),
+                            row(2, "a2", null),
+                            row(3, "a3", "a3_commit")));
+
+            onHudi().executeQuery("ALTER TABLE default." + tableName + " DROP COLUMNS (new_col)");
+            onHudi().executeQuery("INSERT INTO default." + tableName + " VALUES (4, 'a4', 20, 1000)");
+            String afterDropColumn = formatter.format(new Date());
+            assertThat(onTrino().executeQuery("SELECT id, name FROM hudi.default." + tableName + " FOR VERSION AS OF " + afterDropColumn))
```

Comment:
The assertions having `FOR VERSION AS OF` specifies the latest version. That's not a common scenario when using time travel. I would recommend verifying with the past version. Also, it should verify the table definition.